vllm/entrypoints/llm.py

```diff
@@ -548,6 +573,24 @@ def _add_request(
             lora_request=lora_request,
             prompt_adapter_request=prompt_adapter_request)
 
+    def _add_guided_processor(
+            self,
+            params: SamplingParams,
+            guided_options: Optional[GuidedDecodingFields] = None):
+        if guided_options:
+            if guided_options.guided_decoding_backend is None:
+                decoding_config = self.llm_engine.get_decoding_config()
+                guided_options.guided_decoding_backend = (
+                    decoding_config.guided_decoding_backend)
+            guided_logits_processor = get_local_guided_decoding_logits_processor(  #noqa
+                guided_options.guided_decoding_backend, guided_options,
+                self.get_tokenizer())
+            if guided_logits_processor:
+                if params.logits_processors is None:
+                    params.logits_processors = []
+                params.logits_processors.append(guided_logits_processor)
+        return params
```

Comment:
Perhaps we should make this a method of `SamplingParams` instead of repeating the definition in LLM and OpenAI server?

Reply:
I currently want to separate out GuidedDecodingFields, which is used by local LLM, and ChatRequest and CompletetionRequest, used by openai server as they have different ways of retrieving the logits processors. As openai server's retriever of logits processor is async and local LLM is synchronous, it would probably still require separate methods even if it is implement in SamplingParams.
I have implemented all the other suggestions. I really appreciate the time you took to review them! If you have any further suggestions, do let me know!

Reply:
Alright, let's keep them as is for now.